src/nvim/fileio.c

```diff
@@ -5566,9 +5502,13 @@ char *file_pat_to_reg_pat(const char *pat, const char *pat_end, char *allow_dirs
     return xstrdup("^$");
   }
 
+  if (allow_dirs != NULL) {
+    *allow_dirs = false;
```

Comment:
Why is this assignment moved?

Reply:
Woops, really sorry about that, this was a mistake. Hope this didn't cost you too much time.

I have no idea why I did this. It could have been accidental.

Reply:
Oh wait, I didn't realise this was an output variable which wasn't being set due to the early return.

I think as a coding practice, it might be better to require callers to initialize non-NULL pointers used as outputs like this. What do you think?